content/developer/howtos/upgrade/migration_scripts.rst

```diff
@@ -0,0 +1,125 @@
+=================
+Migration scripts
+=================
+
+A migration script is a Python file containing a function called `migrate`, which the upgrade
+process invokes at the appropriate time.
+
+The `migrate` function receives two parameters:
+
+- cr: The current database cursor
+- version: The installed version of the module
+
+Typically, this function executes one or multiple SQL queries and can also access Odoo's ORM, as
+well as the :doc:`../upgrade/upgrade_util`.
+
+As described in :ref:`upgrade_custom/upgraded_database/migrate_data`, you might have to use
+migration scripts to reflect changes from the source code to their corresponding data.
+
+
+.. todo:: Check if needed: How to test them:
+..    Odoo SH : branch in upgrade mode, push a commit
+..    On-premise: receive your dump, run it and upgrade all modules via starting odoo-bin
+
+
+Writing migration scripts
+=========================
+
+To write what we refer as a **custom migration script**, the path of the file should follow this
+template: :file:`<module_name>/migrations/<major_version>.<minor_version>/{pre|post|end}-*.py`.
+
+Follow this steps to create the directory's structure and the Python file:
+
+   #. Create a :file:`migrations` directory inside the custom module.
+   #. Create a :file:`version` directory with the format: :file:`<major_version>.<minor_version>`
+      inside the :file:`migrations` directory.
+
+      - :file:`<major_version>` corresponds to the major version of Odoo (e.g., :file:`16.0` for
+        Odoo 16).
+      - :file:`<minor_version>` corresponds to the upgraded version declared on the module's
+        manifest.
+
+      Migration scripts are only executed when the module is being upgraded. Therefore, the
```

Comment:
```suggestion
      Migration scripts are only executed when the module is being updated. Therefore, the
```

Reply:
I used "upgrade", because that's the name of the option in the interface
![image](https://github.com/odoo/documentation/assets/28193899/0208eba7-54e8-4442-b65a-7ab55b188752)
Although in the command it's "update"
![image](https://github.com/odoo/documentation/assets/28193899/325b070a-c3a9-4fd2-916c-858108155c01)
:zany_face: